app/src/main/java/com/afwsamples/testdpc/common/keyvaluepair/KeyValueBundleArrayFragment.java

```diff
@@ -87,12 +95,37 @@ public static KeyValueBundleArrayFragment newInstance(String key, Bundle[] bundl
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         mKey = getArguments().getString(ARG_KEY);
-        Bundle[] bundles = (Bundle[]) getArguments().getParcelableArray(ARG_INITIAL_VALUE);
+        Object[] bundles = getArguments().getParcelableArray(ARG_INITIAL_VALUE);
         if (bundles == null) {
             bundles = new Bundle[0];
         }
-        mBundleList = new ArrayList<>(Arrays.asList(bundles));
-        mInitialBundleList = new ArrayList<>(mBundleList);
+        mBundleArrayRestrictionEntry = (RestrictionEntry) getArguments().getParcelable(
+                ARG_RESTRICTION_ENTRY);
+
+        if (mBundleArrayRestrictionEntry != null) {
```

Comment:
As I said in previous comment, we have this kind of if else clause all over the codes now.